### ISSUE TITLE
Crashes of [IBGSurveyCache syncSurveysResponse:pausedSurveysIDs:isAnnouncements:]_block_invoke on SDK 10.4.1

### ISSUE THREAD

**Issue by marcofung** [Instabug/Instabug-iOS#410]
```
Crashed: com.apple.main-thread
SIGSEGV 0x0000000b2ef94ce0

Crashed: com.apple.main-thread
0  libobjc.A.dylib                0x19b6c2ed0 objc_retain + 16
1  CoreFoundation                 0x18656eeb4 -[__NSDictionaryM setObject:forKeyedSubscript:] + 388
2  Instabug                       0x103dc8b44 __71-[IBGSurveyCache syncSurveysResponse:pausedSurveysIDs:isAnnouncements:]_block_invoke + 432
3  Instabug                       0x103c95c30 __56-[IBGPromise chainOnQueue:chainedFulfill:chainedReject:]_block_invoke.48 + 52
4  libdispatch.dylib              0x18627c24c _dispatch_call_block_and_release + 32
5  libdispatch.dylib              0x18627ddb0 _dispatch_client_callout + 20
6  libdispatch.dylib              0x18628b800 _dispatch_main_queue_callback_4CF + 920
7  CoreFoundation                 0x18660511c __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
8  CoreFoundation                 0x1865ff120 __CFRunLoopRun + 2508
9  CoreFoundation                 0x1865fe21c CFRunLoopRunSpecific + 600
10 GraphicsServices               0x19e1ca784 GSEventRunModal + 164
11 UIKitCore                      0x18903eee8 -[UIApplication _run] + 1072
12 UIKitCore                      0x18904475c UIApplicationMain + 168
13 9Gag                           0x1024fb604 main + 4333336068 (<compiler-generated>:4333336068)
14 libdyld.dylib                  0x1862be6b0 start + 4
```

### Steps to Reproduce the Problem
Unknown

### Expected Behavior
No crashes

### Actual Behavior
Crashes

### Instabug integration code
Instabug.start(withToken: InstabugService.clientID, invocationEvents: .none)
APM.enabled = true
APM.autoUITraceEnabled = true
CrashReporting.enabled = true
Instabug.reproStepsMode = .disable
BugReporting.enabledAttachmentTypes = [.extraScreenShot, .galleryImage, .screenRecording]

### SDK Version
iOS SDK v10.4.1

### iOS Version
iOS 14

### Device Model
iPhone

**Comment by yousefhamza** [comment]
Hello again @marcofung 👋 
Can you reproduce this crash locally?

**Comment by yousefhamza** [comment]
Can you provide the full stack trace with all threads involved?

**Comment by marcofung** [comment]
@yousefhamza We can't reproduce the issues. It seems related to 
https://github.com/Instabug/Instabug-iOS/issues/406
https://github.com/Instabug/Instabug-iOS/issues/407

**Comment by marcofung** [comment]
We didn't see other threads related to IBG in the crash logs and may not share the whole stacktrace due to organization policy.

**Comment by yousefhamza** [comment]
@marcofung they are in the same module (surveys) but they are not unrelated, we are actively investigating this crash and will get back to you 🙏

**Comment by Kmohamed** [comment]
@marcofung could you share with occurrences count of this crash?

**Comment by marcofung** [comment]
@Kmohamed Our build is currently in phased release, the daily occurrences is ~10 for current user adoption, projected to be 50-100 daily on full rollout.

**Comment by yousefhamza** [comment]
Fixed in 10.4.3
